### ISSUE TITLE
Gizmos broke on WASM

### ISSUE THREAD

**Issue by simbleau** [bevyengine/bevy#14696]
## Bevy version

0.14.1

I have `bevy_gizmos` enabled. The `GizmoPlugin` was added.

## \[Optional\] Relevant system information

MacOS, but running my app on WASM Chromium.

```
Attribute offset (12) with format VertexFormat::Float32x3 (size: 12) doesn't fit in the vertex buffer stride (12).
 - While validating attributes[0].
 - While validating buffers[1].
 - While validating vertex state.
 - While calling [Device].CreateRenderPipeline([RenderPipelineDescriptor "LineGizmo Pipeline 2D"]).
Understand this warning
[Invalid RenderPipeline "LineGizmo Pipeline 2D"] is invalid.
 - While encoding [RenderPassEncoder "main_transparent_pass_2d"].SetPipeline([Invalid RenderPipeline "LineGizmo Pipeline 2D"]).
 - While finishing [CommandEncoder (unlabeled)].
Understand this warning
[Invalid CommandBuffer] is invalid.
 - While calling [Queue].Submit([[Invalid CommandBuffer]])
```

**Comment by alice-i-cecile** [comment]
Is this new in 0.14.1? Can you reproduce on `main`?

**Comment by simbleau** [comment]
> Is this new in 0.14.1? Can you reproduce on `main`?

Don't have a quick enough way to test on main due to countless migration steps

It does reproduce on `=0.14.0`

**Comment by alice-i-cecile** [comment]
Ah, I wasn't asking you to port your app: just try one of the gizmo examples by cloning the repo :) These are basically always universal bugs with weird hardware / driver specific edges.

**Comment by jurisk** [comment]
I ran into this on WASM build when run in the browser, using `bevy_gizmos`, but I haven't isolated it to 0.14.1 vs 0.14.0 or reproduced in a smaller sample (yet).

```
Attribute offset (12) with format VertexFormat::Float32x3 (size: 12) doesn't fit in the vertex buffer stride (12).
 - While validating attributes[0].
 - While validating buffers[1].
 - While validating vertex state.
 - While calling [Device].CreateRenderPipeline([RenderPipelineDescriptor "LineGizmo Pipeline"]).

[Invalid RenderPipeline "LineGizmo Pipeline"] is invalid.
 - While encoding [RenderPassEncoder "main_transparent_pass_3d"].SetPipeline([Invalid RenderPipeline "LineGizmo Pipeline"]).

[Invalid CommandBuffer] is invalid.
 - While calling [Queue].Submit([[CommandBuffer], [CommandBuffer "shadow_pass_command_encoder" from CommandEncoder "shadow_pass_command_encoder"], [CommandBuffer "shadow_pass_command_encoder" from CommandEncoder "shadow_pass_command_encoder"], [CommandBuffer "shadow_pass_command_encoder" from CommandEncoder "shadow_pass_command_encoder"], [CommandBuffer "shadow_pass_command_encoder" from CommandEncoder "shadow_pass_command_encoder"], [CommandBuffer], [CommandBuffer "main_opaque_pass_3d_command_encoder" from CommandEncoder "main_opaque_pass_3d_command_encoder"], [Invalid CommandBuffer]])
```

**Comment by simbleau** [comment]
Ah, yeah. I forgot to mention I'm also running a WASM build.

**Comment by jurisk** [comment]
I suspect that it only appears when:

```rust
gizmos.sphere(position, Quat::default(), 0.1, color);
``` 

...is called.

**Comment by simbleau** [comment]
> I suspect that it only appears when:
> 
> ```rust
> gizmos.sphere(position, Quat::default(), 0.1, color);
> ```
> 
> ...is called.

It's triggered the second I add a call to any gizmo. (I don't have any spheres, I only use 2d rects/lines).

If I disable gizmos (easily done in my app), I don't get this panic.

The panic triggers the first time you render any gizmo for me.

**Comment by jurisk** [comment]
I can repeat this on both `0.14.0` and `0.14.1`, the behaviour appears to be the same.

**Comment by rparrett** [comment]
Could someone do a minimal reproducer? I am not seeing this behavior in any of the deployed wasm examples that utilize gizmos.

m1 max / macOS 14.5 / chrome 127

**Comment by jurisk** [comment]
FWIW:
* https://bevyengine.org/examples-webgpu/gizmos/2d-gizmos/ - fails
* https://bevyengine.org/examples-webgpu/gizmos/3d-gizmos/ - fails
* https://bevyengine.org/examples-webgpu/gizmos/axes/ - works
* https://bevyengine.org/examples-webgpu/gizmos/light-gizmos/ - fails
* https://bevyengine.org/examples/#gizmos - all four WebGL examples work

Tested on two systems:
* Chrome 127.0.6533.100 (Official Build) (64-bit) on Windows 11 Pro.
* Chrome 127 on MacOS 14.5.

**Comment by rparrett** [comment]
Oh, okay, geat. I can reproduce with webgpu too.

**Comment by akimakinai** [comment]
This seems to have been fixed by #14653. Tested on Chrome 127.0.6533.100/Win11, 3d-gizmos example works with e14f3cf402a672adebff9723382eb36076454ca0, while e14f3cf402a672adebff9723382eb36076454ca0^ reproduces the error.

**Comment by akimakinai** [comment]
It seems gizmo joints in WebGPU are still broken on main. #14653 fixed lines but not joints.

```
Attribute offset (12) with format VertexFormat::Float32x3 (size: 12) doesn't fit in the vertex buffer stride (12).
 - While validating attributes[0].
 - While validating buffers[1].
 - While validating vertex state.
 - While calling [Device].CreateRenderPipeline([RenderPipelineDescriptor "LineJointGizmo Pipeline 2D"]).

[Invalid RenderPipeline "LineJointGizmo Pipeline 2D"] is invalid.
 - While encoding [RenderPassEncoder "main_transparent_pass_2d"].SetPipeline([Invalid RenderPipeline "LineJointGizmo Pipeline 2D"]).
```

**Comment by alice-i-cecile** [comment]
Closing as fixed by #14721, but please correct me if this is not the case.
